### PR TITLE
fix errors when missing deployment_name

### PR DESCRIPTION
**Pull request by kkr16**
### Changes:
- Used existing func `DeploymentName()` to validate `deployment_name` is a valid string, and fail early with a user-friendly `DeploymentNameError()` otherwise. If `deployment_name` is:
  - **not set**: `deployment_name must be a string and cannot be empty, cause: deployment_name variable not defined.`
  - **an empty string** `""`: `deployment_name must be a string and cannot be empty, cause: deployment_name was an empty string.`
  - **not a string** (i.e. an integer `100`): `deployment_name must be a string and cannot be empty, cause: deployment_name was not of type string.`
  - **empty**: `deployment_name must be a string and cannot be empty, cause: deployment_name was not of type string.`
-  removed `getDeploymentName()` that returned `undefined` when `deployment_name` was not set. Early error checking renders this obsolete.
- Added a default `deployment_name` to test configs returned from `func getBasicDeploymentConfigWithTestModule()` and `func getDeploymentConfigForTest()` used during unit tests, and updated relevant unit tests accordingly

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: ghpc.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    https://www.apache.org/licenses/LICENSE-2.0
```
Comment (heyealex) [inline ghpc.go:8]: Was this change intentional? It appears this line is 4 spaces in other files, it's better to keep it consistent.

File: pkg/config/config.go [hunk 2]
```diff
@@ -519,8 +523,8 @@ func ConvertMapToCty(iMap map[string]interface{}) (map[string]cty.Value, error)
 // global variables, then they are replaced by the cty.Value of the
 // corresponding entry in the origin. All other cty.Values are unmodified.
 // ERROR: if (somehow) the cty.String cannot be converted to a Go string
-// ERROR: rely on HCL TraverseAbs to bubble up "diagnostics" when the global variable
-//        being resolved does not exist in b.Vars
+// ERROR: rely on HCL TraverseAbs to bubble up "diagnostics" when the global
+// variable being resolved does not exist in b.Vars
```
Comment (heyealex) [inline pkg/config/config.go:527]: I found this more clear before, though I'm not opposed to the change. Was this made by the go linter?

**Review by heyealex — CHANGES_REQUESTED** [review]
Thanks Karim, this is great! I just had a couple comments related to some formatting changes that might have been an issue with to the go linter.

File: ghpc.go (hunk 1 above)
Comment (kkr16) [reply ghpc.go]: Not intentional. Change was proposed by the linter, and I just went with it. 
I reverted it for now.

File: pkg/config/config.go (hunk 2 above)
Comment (kkr16) [reply pkg/config/config.go]: A worse change change was proposed by the linter, and this is how I was able to appease the linter. 
I reverted it for now.

**Review by heyealex — DISMISSED** [review]

**Review by heyealex — APPROVED** [review]
